spec/var-op-bash.test.sh

```diff
@@ -300,10 +300,11 @@ status=0
 status=0
 ## END
 ## OK osh STDOUT:
-status=1
+a b c
+status=0
 a $'b\\nc'
 status=0
-a
+a a a
```

Comment:
I can see that OSH differs from bash because of a quoting difference, which is OK

But it also seems to differ with `a a a` vs. `''` ?

Or maybe that is addressed in a follow-up?


TBH I don't understand the reasons for the evaluation logic - in these cases, my review falls back to "does OSH match bash?"

Reply:
Ah! Thanks for pointing out this. This is just my oversight. OSH should generate the same result as Bash. Let me think about how we should distinguish array-origin BashArray from `$@`-origin BashArray.

Reply:
It is OK if we fix that later (or even not at all if nobody runs into it, and bash does not document it)

Sometimes when I encounter a quirk like this, I split it into a separate test case, and then mark it `oils_failures_allowed += 1`


(Basically because there seem to be infinite bash implementation details!   I think using `@` with a type query is EXTREMELY obscure)

So that way the first test case have bash STDOUT === osh STDOUT, which is a nice proprety

Reply:
Could you wait for a while? If it turns out to be easy to fix, I can fix it within this PR. If not, I'll handle it in a separate PR later.

Reply:
I decided to defer it. The best solution could be to introduce a new type to represent the list coming from `$@`, but one might relate it to the forthcoming switching from BashArray to SparseArray. We can discuss that later.

Reply:
Now this is an allowed OSH failure: fda3cd43d785c0c2d71ff9084091ee5ee63555d2